Case Study 01: Sales Force Training / Muhammad Fakhri Azhar

In [4]:
#Import Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

In [7]:
#Create DataFrame
data = [100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65]
TransactionAmount = pd.DataFrame(data, columns=['TransactionAmount'])
TransactionAmount

TransactionAmount
0                 100
1                 150
2                  50
3                 100
4                 130
5                 120
6                 100
7                  85
8                  70
9                 150
10                150
11                120
12                 50
13                100
14                100
15                140
16                 90
17                150
18                 50
19                 90
20                120
21                100
22                110
23                 75
24                 65

In [10]:
#Central Tendency
import statistics

TransactionAmount = TransactionAmount['TransactionAmount'].tolist()
print('Mean: ',statistics.mean(TransactionAmount))
print('Median: ',statistics.median(TransactionAmount))
print('Mode: ',statistics.mode(TransactionAmount))

Mean:  102.6
Median:  100
Mode:  100


Analisis Central Tendency:
1. Rata-rata penjualan naik menjadi `$102.6` (lebih tinggi dari rata-rata sebelumnya `$100`), menunjukkan adanya peningkatan setelah pelatihan.
2. Median tetap di `$100`, yang berarti setengah dari pekerja masih memiliki penjualan di bawah atau sama dengan `$100`. Ini mengindikasikan bahwa peningkatan rata-rata mungkin dipengaruhi oleh beberapa pekerja dengan penjualan tinggi (outlier).
3. Mode = $100, menunjukkan bahwa jumlah terbesar pekerja masih mencatat penjualan sekitar angka ini.

In [18]:
#Variability
import numpy as np
print('Variance: ', np.var(TransactionAmount))
print('Std Deviasi: ', np.std(TransactionAmount))
print('Range: ', np.max(TransactionAmount) - np.min(TransactionAmount))
print('Quartile 1: ', np.quantile(TransactionAmount, 0.25))
print('Quartile 2: ', np.quantile(TransactionAmount, 0.5))
print('Quartile 3: ', np.quantile(TransactionAmount, 0.75))

Variance:  972.2399999999999
Std Deviasi:  31.18076330047101
Range:  100
Quartile 1:  85.0
Quartile 2:  100.0
Quartile 3:  120.0


Analisis Variabilitas Penjualan Setelah Pelatihan :
1. Standar deviasi (31.18) dan varians (972.24) cukup besar, menunjukkan perbedaan penjualan antar pekerja cukup tinggi.
2. Range (100) menunjukkan selisih besar antara penjualan tertinggi dan terendah.
3. Kuartil (Q1 = 85, Q2 = 100, Q3 = 120) → Mayoritas pekerja memiliki penjualan antara 85 - 120.
4. Variasi yang besar bisa disebabkan oleh beberapa pekerja dengan penjualan tinggi (outlier).

In [23]:
#Define H0 and H1
#H0: rata-rata sebelum pelatihan, transactions amount = 100
#H1: rata-rata setelah pelatihan, transactions amount > 100

In [26]:
#Define alpha = 0.05

In [29]:
#Calculate t statistics and P-value
#Perform one sample t-test

statistic, p_value = stats.ttest_1samp(a=TransactionAmount, popmean=100)
df = len(TransactionAmount) - 1

print(f'T-Statistic: {statistic}')
print(f'P-Value: {p_value}')
print(f'Degree of Freedom: {df}')

alpha = 0.05 
if p_value < alpha:
    print("Tolak H0, rata-rata berbeda secara signifikan dari $100.")
else:
    print("Gagal tolak H0, tidak ada perbedaan signifikan dengan $100.")

T-Statistic: 0.4085001556802841
P-Value: 0.6865284813438117
Degree of Freedom: 24
Gagal tolak H0, tidak ada perbedaan signifikan dengan $100.


In [32]:
#statistic= 0.4085001556802841
#pvalue= 0.6865284813438117
#Degree of Freedom = 24

Conclude with alpha and p-value:
P Value memiliki nilai lebih besar dari alpha (0,68 > 0,05), maka perusahaan x Gagal menolak H0, jadi tidak ada perbedaan yang signifikan dengan $100.
Conclusion: 
Tidak ada perbedaan signifikan antara rata-rata penjualan setelah pelatihan (`$102.6`) dengan sebelum (`$100`).

In [37]:
#Conclude with t statistics and critical region
#Find the critical region

In [40]:
#H1: larger (right side)
crit_region = stats.t.ppf(1-alpha, df)
print(f'Critical Region: {crit_region}')

Critical Region: 1.7108820799094275


Critical region result
t-stat < dari critical region, maka perusahaan x gagal menolak H0
Conclusion: Tidak ada cukup bukti untuk menyatakan bahwa rata-rata sampel berbeda secara signifikan dari rata-rata populasi yang diuji. Dengan kata lain, tidak ada efek yang signifikan dari pelatihan yang dilakukan oleh perusahaan x. 